test/puppetlabs/trapperkeeper/services/status/status_service_test.clj

```diff
@@ -176,3 +176,46 @@
                 "message" (str "No status function with version 3 "
                             "found for service foo")}
               (json/parse-string (slurp (:body resp)))))))))
+
+(defn response->status
+  [resp]
+  (:status (json/parse-string (slurp (:body resp)) true)))
+
+(deftest level>=test
+  (testing "use of level>= to implement a status function"
+    (let [my-status (fn [level]
+                      (let [level>= (partial status-core/level>= level)]
+                        {:is-running true
+                         :status (cond-> {:this-is-critical "foo"}
+                                   (level>= :info) (assoc :bar "bar"
+                                                          :baz "baz")
+                                   (level>= :debug) (assoc :x "x"
+                                                           :y "y"
+                                                           :z "y"))}))
```

Comment:
I'm OK with this; it's better than what we had before and it's better than what I had in my pseudo-code.

That said, it might be nice to separate out the ">=" logic so that the user could compose whatever fns they wanted?

So, your partial could look like this:

```clj
(let [level>= (partial >= (status-core/numeric-level level)]
```

That would be a bit more flexible, and from the user perspective the rest of your example code could stay exactly as it is... and we could include an example like yours in the docs?  If you are on board with that, I'm not necessarily in love with the name `numeric-level`... could be like `level-value` or something.